interfaces/builtin/raw_input.go

```diff
@@ -0,0 +1,77 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const rawInputSummary = `allows access to raw input devices`
+
+const rawInputBaseDeclarationSlots = `
+  raw-input:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const rawInputConnectedPlugAppArmor = `
+# Description: Allow reading and writing to raw input devices
+
+/dev/input/* rw,
+
+# Allow reading for supported event reports for all input devices. See
+# https://www.kernel.org/doc/Documentation/input/event-codes.txt
+/sys/devices/**/input[0-9]*/capabilities/* r,
+
+# For using udev
+/run/udev/data/c13:[0-9]* r,
+/run/udev/data/+input:input[0-9]* r,
+`
+
+var rawInputConnectedPlugUDev = []string{
+	`KERNEL=="event[0-9]*", SUBSYSTEM=="input"`,
+	`KERNEL=="mice"`,
+	`KERNEL=="mouse[0-9]*"`,
+	`KERNEL=="ts[0-9]*"`,
```

Comment:
touch screen? I see this in x11, wayland and mir interfaces but no comments there.

Reply:
Yep.